Tests/AnalyticsTests/Tests.swift

```diff
@@ -27,6 +51,22 @@ final class Tests: XCTestCase {
         XCTAssertEqual(observer.eventName, ViewEvent.view.name)
         XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
     }
+    
+    func testLogEventAppendSubsequentLogValues() {
+        let action = AnalyticsAction(values: .init())
+        var values = AnalyticsValues()
+        var newValues = AnalyticsValues()
+        
+        values[keyPath: \.source] = .contactList
+        action(.view, appending: values)
+        XCTAssertEqual(observer.eventName, ViewEvent.view.name)
+        XCTAssertEqual(observer.params, [SourceAnalyticsKey.key: Source.contactList.rawValue])
+        
+        newValues[keyPath: \.component] = .button
+        action(.view, appending: newValues)
```

Comment:
I think the test is right but it's hard to read cause you're introducing essentially 3 values. The first you `.init` then the followings `var`'s.

I think an improvement would be init the action with the first one. 

Then call the `appending` method.

Then test that both values are there.

Next call the log without change values.

Finally test that only the original value is there.

Should be easier to understand the test. Super important in all code but especially knowing what's being tested.

Also I think test function names aren't great. Should reconsider them.